openverse_catalog/dags/providers/provider_api_scripts/museum_victoria.py

```diff
@@ -1,203 +1,175 @@
-            image_count = image_store.add_item(
-                foreign_identifier=img.get("image_id"),
-                foreign_landing_url=foreign_landing_url,
-                image_url=img.get("image_url"),
-                height=img.get("height"),
-                width=img.get("width"),
-                license_info=license_info,
-                title=title,
-                creator=img.get("creators"),
-                meta_data=meta_data,
-            )
-    return image_count
-
-
-def _get_media_info(media_data):
-    image_data = []
-    for media in media_data:
-        media_type = media.get("type")
-        if media_type == "image":
-            image_id = media.get("id")
-            image_url, height, width = _get_image_data(media)
-            license_url = _get_license_url(media)
-            if image_url is None or image_id is None or license_url is None:
+class ImageDetails(TypedDict, total=False):
+    foreign_identifier: str
+    image_url: str
+    license_info: LicenseInfo
+    foreign_landing_url: str
+    title: str
+    meta_data: Dict
+    height: int | None
+    width: int | None
+    creator: str | None
+
+
+class VictoriaDataIngester(ProviderDataIngester):
+    providers = {"image": prov.VICTORIA_DEFAULT_PROVIDER}
+    endpoint = "https://collections.museumsvictoria.com.au/api/search"
+    headers = {"User-Agent": prov.UA_STRING, "Accept": "application/json"}
+    batch_limit = 100
+    delay = 5
+    # This set is used to prevent duplicate images of the same items
+    RECORDS_IDS = set()
+    LANDING_PAGE = "https://collections.museumsvictoria.com.au/"
+    LICENSE_LIST = [
+        "public domain",
+        "cc by",
+        "cc by-nc",
+        "cc by-nc-sa",
+        "cc by-nc-nd",
+        "cc by-sa",
+    ]
+
+    def ingest_records(self, **kwargs):
+        for license_ in VictoriaDataIngester.LICENSE_LIST:
+            super().ingest_records(license_=license_)
+
+    def get_batch_data(self, response_json):
+        if response_json:
+            return response_json
+        return None
+
+    def get_next_query_params(self, old_query_params: Optional[Dict], **kwargs) -> Dict:
+        if not old_query_params:
+            return {
+                "hasimages": "yes",
+                "perpage": self.batch_limit,
+                "imagelicense": kwargs["license_"],
+                "page": 0,
+            }
+        else:
+            return {
+                **old_query_params,
+                "page": old_query_params["page"] + 1,
+            }
+
+    def get_record_data(self, data: Dict):
+        object_id = data.get("id")
+        if object_id in self.RECORDS_IDS:
+            return None
+        self.RECORDS_IDS.add(object_id)
+        foreign_landing_url = f"{self.LANDING_PAGE}{object_id}"
+
+        if (media_data := data.get("media")) is None:
+            return None
+        images = VictoriaDataIngester._get_images(media_data)
```

Comment:
This doesn't matter too much but all the static reference calls on `VictoriaDataIngester` can be replaced with `self` (it at least makes them a bit shorter for formatting haha)

Reply:
Much better (shorter) with `self` :)